kbase-extension/static/kbase/js/widgets/function_output/kbaseReportView.js

```diff
@@ -233,155 +216,108 @@ define([
             };
         },
 
-        makeIframeSrcDataPlain: function (arg) {
-            const t = html.tag,
-                iframe = t('iframe');
-
-            const iframeId = 'frame_' + html.genId();
-
-            // The iframe content needs requirejs amd.
-
-            const width = arg.width || '100%',
-                iframeContent = arg.content,
-                iframeHtml = iframe({
-                    style: {
-                        display: 'block',
-                        width: width,
-                        height: arg.height || 'auto',
-                        margin: 0,
-                        padding: 0,
-                    },
-                    dataFrame: iframeId,
-                    frameborder: '0',
-                    id: iframeId,
-                    src: 'data:text/html;charset=utf-8,' + encodeURIComponent(iframeContent),
-                });
-
-            return {
+        /**
+         *
+         * @param {Object} arg - properties are:
+         *  - height - height of the iframe, default = 'auto'
+         *  - src - source url of the iframe (either this OR content should be used)
+         *  - content - raw content to put in the iframe, will be encoded (this OR src should be used)
+         */
+        makeIframeFromSrc: function (arg) {
```

Comment:
Combined `makeIframeSrcDataPlain` and `makeIfromSrc` into this one function, as they only differed by a single line.

Reply:
`makeIframe` is complex enough on its own to warrant being left as its own function.